packages/yarnpkg-pnp/sources/loader/makeManager.ts

```diff
@@ -132,14 +132,16 @@ export function makeManager(pnpapi: PnpApi, opts: MakeManagerOptions) {
       if (cached !== undefined)
         return addToCacheAndReturn(start, curr, cached);
 
-      const candidate = ppath.join(curr, `.pnp.js` as Filename);
-      if (xfs.existsSync(candidate) && xfs.statSync(candidate).isFile())
-        return addToCacheAndReturn(start, curr, candidate);
-
-      const cjsCandidate = ppath.join(curr, `.pnp.cjs` as Filename);
+      const cjsCandidate = ppath.join(curr, Filename.pnpCjs);
       if (xfs.existsSync(cjsCandidate) && xfs.statSync(cjsCandidate).isFile())
         return addToCacheAndReturn(start, curr, cjsCandidate);
 
+      // We still support .pnp.js files to improve multi-project compatibility.
+      // TODO: Remove support for .pnp.js files after they stop being used.
+      const legacyCjsCandidate = ppath.join(curr, Filename.pnpJs);
+      if (xfs.existsSync(legacyCjsCandidate) && xfs.statSync(legacyCjsCandidate).isFile())
+        return addToCacheAndReturn(start, curr, legacyCjsCandidate);
```

Comment:
```suggestion
      const legacyJsCandidate = ppath.join(curr, Filename.pnpJs);
      if (xfs.existsSync(legacyJsCandidate) && xfs.statSync(legacyJsCandidate).isFile())
        return addToCacheAndReturn(start, curr, legacyJsCandidate);
```

Reply:
🤔 I initially called it this way because I only wanted to make the distinction between CJS and ESM (which you're implementing in #2161) as formats, not as extensions. I'm not sure if calling it `legacyJsCandidate` & `jsLegacy` is better than `legacyCjsCandidate` & `cjsLegacy`. What do you think?

Reply:
Ah, in that case perhaps just `jsCandidate` and `cjsCandiate`